contracts/FeralfileVaultV2.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+import {FeralfileSaleDataV2} from "./FeralfileSaleDataV2.sol";
+import {ECDSASigner} from "./ECDSASigner.sol";
+
+contract FeralfileVaultV2 is Ownable, FeralfileSaleDataV2, ECDSASigner {
+    mapping(bytes32 => bool) private _paidSale;
+
+    constructor(address signer_) ECDSASigner(signer_) {}
+
+    /// @notice pay for buyArtwork to a FFV4 contract destination.
+    /// @param r_ - part of signature for validating parameters integrity
+    /// @param s_ - part of signature for validating parameters integrity
+    /// @param v_ - part of signature for validating parameters integrity
+    /// @param saleData_ - the sale data
+    function payForSaleV2(
```

Comment:
Should be `payForSale`

Reply:
the payload is different from the old Vault, so I changed the name to reduce the mistake when calling.

Reply:
Unless it makes us confused, otherwise i dont see any reason to name it with the version suffix. 
This function belongs to separate contract `FeralfileVaultV2` that actually doesn't make us confused when using it.